### DISCUSSION TITLE
Mouse cursor, small arrow cursor with a yellow border has no shadow, and wrong cursor

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#2772] [Ideas]
In the original game, for the adventure map and screens, the small arrow cursor with a yellow border has a shadow.
This is not the case in fheroes2.

In original game:
![Cursor type and shadow 001](https://user-images.githubusercontent.com/43583994/107424269-c0ee5300-6b1d-11eb-86b6-1cc88b8dd2d0.png)
![Cursor type and shadow 002](https://user-images.githubusercontent.com/43583994/107424287-c481da00-6b1d-11eb-936b-233cc0526c32.png)
![Cursor type and shadow 003](https://user-images.githubusercontent.com/43583994/107424298-c8156100-6b1d-11eb-8377-6484f343c499.png)

In fheroes2:
![Cursor type and shadow 008](https://user-images.githubusercontent.com/43583994/107424399-dfece500-6b1d-11eb-8190-572c6018859d.png)
![Cursor type and shadow 009](https://user-images.githubusercontent.com/43583994/107424408-e4190280-6b1d-11eb-8bb9-b4f9a77afd6a.png)
![Cursor type and shadow 010](https://user-images.githubusercontent.com/43583994/107424415-e7ac8980-6b1d-11eb-9be3-91869f708bc6.png)

In the original game, during the combat, a small arrow cursor without yellow border more discreet and without shadow is used:
![Cursor type and shadow 004](https://user-images.githubusercontent.com/43583994/107424324-cc417e80-6b1d-11eb-9d0e-269d27a41ce7.png)

In the original game, during the combat, when a window is open, the small arrow cursor with a yellow border with a shadow is used:
![Cursor type and shadow 006](https://user-images.githubusercontent.com/43583994/107424355-d4012300-6b1d-11eb-8c9b-f48bb24f4e1e.png)

Except for some windows where the small arrow cursor without a yellow border more discreet and without shadow is used:
![Cursor type and shadow 005](https://user-images.githubusercontent.com/43583994/107424336-cfd50580-6b1d-11eb-8514-59db3ff072ba.png)
![Cursor type and shadow 007](https://user-images.githubusercontent.com/43583994/107424370-d82d4080-6b1d-11eb-9f84-f12f68fbb7d7.png)

In fheroes2, during the combat, a small arrow cursor without yellow border more discreet and without shadow is used, all right:
![Cursor type and shadow 011](https://user-images.githubusercontent.com/43583994/107424434-ebd8a700-6b1d-11eb-94e9-ee4dc6740f51.png)

In fheroes2, during the combat, when a window is open, the small arrow cursor with a yellow border is used, the shadow is missing:
![Cursor type and shadow 014](https://user-images.githubusercontent.com/43583994/107424474-f6933c00-6b1d-11eb-8142-11816e348f7e.png)

In fheroes2, for other windows, it is not the same cursor as in the original game that is used:
![Cursor type and shadow 012](https://user-images.githubusercontent.com/43583994/107424447-eed39780-6b1d-11eb-9f6e-40e49d4463ba.png)
![Cursor type and shadow 013](https://user-images.githubusercontent.com/43583994/107424461-f2ffb500-6b1d-11eb-9b70-48205a37f053.png)

I suggest standardizing this a little bit:
- The small arrow cursors with and without yellow border should always have a shadow.
- The small arrow cursor with a yellow border is used everywhere except during combat.
- During combat, when a window is open, the small arrow cursor with a yellow border is used.

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu !
Cursor missing shadow for the one with yellow border is SDL2 related problem.
Could you please make a new issue for it?
On subject:
I really don't understand, why this game has two these types of cursors. What is the purpose?
Between, I really don't care, where would be one or another cursor. Your proposal, that non-bordered cursor should appear only during battle looks logical. I've no objections.
Are you sure, that shadow added for non-bordered cursor will look good?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Hi, @LeHerosInconnu !
> Cursor missing shadow for the one with yellow border is SDL2 related problem.
> Could you please make a new issue for it?

Done here: https://github.com/ihhub/fheroes2/issues/2810.

> On subject:
> I really don't understand, why this game has two these types of cursors. What is the purpose?

During the combat, the small arrow-shaped cursor has no yellow border to be more discreet.
The cursor disturbs as little as possible the visual aspect of the battlefield (the beauty of the game).
During the battle, when a window is open, the player must interact in that window.
The arrow-shaped cursor with a yellow border is then used to make it easier to visually locate his position on the screen.
(When the player has to interact on the battlefield, the possible actions have specific cursors (casting a spell, etc.)).

> Between, I really don't care, where would be one or another cursor. Your proposal, that non-bordered cursor should appear only during battle looks logical. I've no objections.
> Are you sure, that shadow added for non-bordered cursor will look good?

Ideally, during the combat, the small arrow-shaped cursor with a yellow border should be used when the mouse position is located in the area of the information bar and all buttons at the bottom of the screen, to allow the player to discern it more easily as the player then enters a phase of interaction with this part of the interface.
The shadow would already be a plus to improve the perception of the mouse cursor without yellow border in this area.
(And the cursor with a yellow border should have a shadow anyway.)

**Reply by Branikolog** [reply]
> During the combat, the small arrow-shaped cursor has no yellow border to be more discreet.

> The shadow would already be a plus to improve the perception of the mouse cursor without yellow border in this area.

It seems to me, these both statements are opposite to each other... )

Me personally, like the cursor without yellow border, but I agree, it is much less noticeable. Especially over a battlefield.
Changing the cursor to a yellow-bordered if we point bottom log and buttons seems logical, but a little bit confusing and not aesthetically pleasing, if we notice it changing on hovering that areas of UI and battlefield.)

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > During the combat, the small arrow-shaped cursor has no yellow border to be more discreet.
> 
> > The shadow would already be a plus to improve the perception of the mouse cursor without yellow border in this area.
> 
> It seems to me, these both statements are opposite to each other... )

No, in the first case, the player will observe what is happening on the screen (creatures moving, etc.) but he still has the possibility to prepare his next action with the more discreet cursor.

(The program can't determine whether the player will simply watch the action on the battlefield and perform his next action afterwards (without preparing it in advance), or prepare his next action while the action is being performed on the battlefield (two different types of players), so the discreet cursor is used to allow both possibilities to different types of players.)

In the second case, all types of players will interact with the bottom of the interface (outside the battlefield), so the cursor should be more visible because seeing the cursor position is more important here.

> Me personally, like the cursor without yellow border, but I agree, it is much less noticeable. Especially over a battlefield.
> Changing the cursor to a yellow-bordered if we point bottom log and buttons seems logical, but a little bit confusing and not aesthetically pleasing, if we notice it changing on hovering that areas of UI and battlefield.)

Cursor visibility is important here, and the shadow also brings a plus on the aesthetic side.

**Reply by Branikolog** [reply]
Talking about aesthetic, I mean that looks quite strange for me, if I notice that cursor changes while waving over log and returning back to battlefield... I feel such cursor behaviour quite alien for Heroes 2. 
My point: it shouldn't change within single screen interaction area. Only, if we press something and open new interaction dialogs...
But anyway, it's just my point of view... We I want to listen more opinions on this question...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Talking about aesthetic, I mean that looks quite strange for me, if I notice that cursor changes while waving over log and returning back to battlefield... I feel such cursor behaviour quite alien for Heroes 2.
> My point: it shouldn't change within single screen interaction area. Only, if we press something and open new interaction dialogs...
> But anyway, it's just my point of view... We I want to listen more opinions on this question...

The two cursors are graphically very similar, and at the same time this small change can be eye-catching, which is also the intended goal here.

**Reply by Branikolog** [reply]
I like your arguments. 
As well as find mine logical and not bad at all.
And again, if other people would love your proposal more, I won't resist making in such way.
Anyway, I think, adding a shadow for non-bordered cursor could be a good "improvement" idea... If it will look good. (I'm afraid, that dark, non-bordered cursor will look illegible and blurred because of shadow.)

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> I like your arguments.
> As well as find mine logical and not bad at all.
> And again, if other people would love your proposal more, I won't resist making in such way.
> Anyway, I think, adding a shadow for non-bordered cursor could be a good "improvement" idea... If it will look good. (I'm afraid, that dark, non-bordered cursor will look illegible and blurred because of shadow.)

Here the cursor without a yellow border and with a shadow:

![Ecran avec curseur sans bord jaune et ombre 01](https://user-images.githubusercontent.com/43583994/108285431-5c6b6d80-7187-11eb-9779-6454e9e09e3c.png)

I would prefer the cursor with a yellow border in this area which would be even more visible. :)

**Reply by Branikolog** [reply]
> Here the cursor without a yellow border and with a shadow:

I'm really curious, how could you manage to create your representative screenshots? :)

> I'm afraid, that dark, non-bordered cursor will look illegible and blurred because of shadow.

I mean, over the battlefield. Over the bright, plain GUI elements it will definitely look nice (And your screenshot shows it perfectly!)

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> > Here the cursor without a yellow border and with a shadow:
> 
> I'm really curious, how could you manage to create your representative screenshots? :)

When I raised my mouse slightly over the desk, a shadow appeared. :D
No, I drew the shadow in an image editor by counting the pixels. :)

> > I'm afraid, that dark, non-bordered cursor will look illegible and blurred because of shadow.
> 
> I mean, over the battlefield. Over the bright, plain GUI elements it will definitely look nice (And your screenshot shows it perfectly!)

In the battlefield area of the screen it doesn't matter as much, because in this part of the screen the cursor has to remain discreet.

This is what the cursor would look like in the battlefield area:

![Ecran avec deux curseurs sans bord jaune et ombre 03](https://user-images.githubusercontent.com/43583994/108420537-3bad2180-7234-11eb-8ca8-4de4eb682a27.png)

**Comment by LeHerosInconnu** [comment]
It's okay the way it works currently.
I close the discussion.
